pkg/ruler/ruler_test.go

```diff
@@ -844,9 +899,90 @@ func TestGetRules(t *testing.T) {
 				time.Sleep(100 * time.Millisecond)
 			}
 
+			scrape := func(rID string, scrapeCh chan prometheus.Metric, doneCh chan struct{}) {
+				go func() {
+					tick := time.NewTicker(10 * time.Millisecond)
+					for {
+						select {
+						case <-tick.C:
+							if len(scrapeCh) >= 100 {
+								continue
+							}
+							m := rulerManagerMap[rID]
+							data := m.userManagerMetrics.regs.BuildMetricFamiliesPerUser()
+							data.SendSumOfCountersPerUserWithLabels(scrapeCh, m.userManagerMetrics.IterationsScheduled, "prometheus_rule_group_iterations_total", "rule_group")
+						case <-doneCh:
+							return
+						}
+					}
+				}()
+			}
+
+			waitUntil := func(rID string, user string, ruleGroupList rulespb.RuleGroupList) {
```

Comment:
I think this logic looks very complicated 😰 , why are we using this complicated approach instead of simply putting in a sleep equivalent to 4-5 eval interval ?